README.md

```diff
@@ -0,0 +1,73 @@
+# dnsprove.js 
+
+## Functionalities
+
+- Fetches DNS information of given domain name and type
+- Validates DNS reponse and constructs proofs
+- Submits the proofs into DSNSEC Oracle smart contract
+- Works from both browser and from node.js
+
+## Installing
+
+```
+npm install '@ensdomains/dnsprovejs' --save
+```
+
+## Usage
+
+```js
+var Web3      = require('web3');
+var provider  = new Web3.providers.HttpProvider();
+var DnsProve  = require('dnsprove');
+var dnsprove  = new DnsProve(provider);
+dnsResult.found // returns true/false
+var dnsResult = await dnsprove.lookup('TXT', '_ens.matoken.xyz');
+var oracle    = await dnsprove.getOracle('0x123...');
+assert(dnsResult.found);
+var proofs = dnsResult.proofs;
+for(i = 0; i < proofs.length; i++){
+  var proof = proofs[i];
+  // proof.rrsig
+  // proof.signature
+  if(!await oracle.knownProof(proof)){
+    await oracle.submitProof(proof, proofs[i-1], {from:address})
```

Comment:
Won't this fail on iteration 0 of the loop?

Reply:
It doesn't fail. If `proofs[i-1]` is `null` it takes the previous proof from anchor.